extensions/libraries/redcore/api/api.php

```diff
@@ -199,9 +199,27 @@ public static function getPostedData()
 				}
 			}
 
-			// We try to transform it into XML
-			if (is_null($parsedData) && $xml = @simplexml_load_string($inputData))
+			$xml = @simplexml_load_string($inputData);
+
+			// We try to transform soap XML
+			if (is_null($parsedData) && $xml !== false)
 			{
+				/*
+					XML Namespaces suck and I couldn't get it to read the data accurately for SOAP
+					So I'm stripping everything between < and : in the xml and re-rendering it
+				    I know. I know. My skin is crawling too, but its the best I could do at the moment.
+				    If someone can make this work with namespaces that would be B-E-A-UTIFUL!
+				*/
+				$cleanData = preg_replace('/<(\/)?([a-z0-9]+):([a-z0-9]+)/i', '<$1$3', $inputData);
+				$xml = @simplexml_load_string($cleanData);
```

Comment:
I won't pretend to know what the soap server is doing back there and I'm not to familiar with RPC, but in order to set the language to JInput, we need to get those variables out of the xml. 

Since this method seemed like it was designed to do that, I just assumed making happen here would be the best place. I can move it somewhere else if you'd like, just let me know where.